operator/builtin/parser/keyvalue/keyvalue.go

```diff
@@ -0,0 +1,167 @@
+)
+
+func init() {
+	operator.Register("key_value_parser", func() operator.Builder { return NewKVParserConfig("") })
+}
+
+// NewKVParserConfig creates a new key value parser config with default values
+func NewKVParserConfig(operatorID string) *KVParserConfig {
+	return &KVParserConfig{
+		ParserConfig:  helper.NewParserConfig(operatorID, "key_value_parser"),
+		Delimiter:     "=",
+		PairDelimiter: "",
+	}
+}
+
+// KVParserConfig is the configuration of a key value parser operator.
+type KVParserConfig struct {
+	helper.ParserConfig `mapstructure:",squash" yaml:",inline"`
+
+	Delimiter     string `mapstructure:"delimiter" yaml:"delimiter"`
+	PairDelimiter string `mapstructure:"pair_delimiter" yaml:"pair_delimiter"`
+}
+
+// Build will build a key value parser operator.
+func (c KVParserConfig) Build(context operator.BuildContext) ([]operator.Operator, error) {
+	parserOperator, err := c.ParserConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Delimiter == c.PairDelimiter {
+		return nil, errors.New("delimiter and pair_delimiter cannot be the same value")
+	}
+
+	if len(c.Delimiter) == 0 {
+		return nil, errors.New("delimiter is a required parameter")
+	}
+
+	// split on whitespace by default, if pair delimiter is set, use
+	// strings.Split()
+	pairSplitFunc := splitStringByWhitespace
+	if c.PairDelimiter != "" {
+		pairSplitFunc = func(input string) []string {
+			return strings.Split(input, c.PairDelimiter)
+		}
+	}
+
+	kvParser := &KVParser{
+		ParserOperator: parserOperator,
+		delimiter:      c.Delimiter,
+		pairSplitFunc:  pairSplitFunc,
+	}
+
+	return []operator.Operator{kvParser}, nil
+}
+
+// KVParser is an operator that parses key value pairs.
+type KVParser struct {
+	helper.ParserOperator
+	delimiter     string
+	pairSplitFunc func(input string) []string
+}
+
+// Process will parse an entry for key value pairs.
+func (kv *KVParser) Process(ctx context.Context, entry *entry.Entry) error {
+	return kv.ParserOperator.ProcessWith(ctx, entry, kv.parse)
+}
+
+// parse will parse a value as key values.
+func (kv *KVParser) parse(value interface{}) (interface{}, error) {
+	switch m := value.(type) {
+	case string:
+		return kv.parser(m, kv.delimiter)
+	default:
+		return nil, fmt.Errorf("type %T cannot be parsed as key value pairs", value)
+	}
+}
+
+func (kv *KVParser) parser(input string, delimiter string) (map[string]interface{}, error) {
+	if len(input) == 0 {
+		return nil, fmt.Errorf("parse from field %s is empty", kv.ParseFrom.String())
+	}
+
+	parsed := make(map[string]interface{})
+
+	var err error
+	for _, raw := range kv.pairSplitFunc(input) {
+		m := strings.Split(raw, delimiter)
+		if len(m) != 2 {
+			e := fmt.Errorf("expected '%s' to split by '%s' into two items, got %d", raw, delimiter, len(m))
+			err = multierror.Append(err, e)
+			continue
+		}
+
+		key := cleanString(m[0])
+		value := cleanString(m[1])
+
+		// TODO: Check if key already exists and fail if so?
```

Comment:
Failing entirely seems harsh. Probably a lot better to get a partial log than none. 

We could alternately log the error. Not sure the performance hit of checking every time is worth it for the rare case when it matters though.

Up to you, but we should decide and remove the TODO.